<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>894</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>1</horstretch>
    <verstretch>1</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>500</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>CryptoChat Client</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
     <horstretch>1</horstretch>
     <verstretch>1</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>500</width>
     <height>330</height>
    </size>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <item>
     <widget class="QWidget" name="messagingWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_6" stretch="0,0">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QTableWidget" name="messagesTableWidget">
         <property name="autoFillBackground">
          <bool>false</bool>
         </property>
         <property name="sizeAdjustPolicy">
          <enum>QAbstractScrollArea::AdjustIgnored</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="horizontalScrollMode">
          <enum>QAbstractItemView::ScrollPerPixel</enum>
         </property>
         <property name="showGrid">
          <bool>false</bool>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
         <property name="columnCount">
          <number>2</number>
         </property>
         <attribute name="horizontalHeaderCascadingSectionResizes">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="horizontalHeaderStretchLastSection">
          <bool>true</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderDefaultSectionSize">
          <number>20</number>
         </attribute>
         <attribute name="verticalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string>Sender</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Message</string>
          </property>
         </column>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="messageLineEdit">
         <property name="placeholderText">
          <string>Type Message...</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="contactsWidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QToolButton" name="addContactButton">
         <property name="minimumSize">
          <size>
           <width>24</width>
           <height>24</height>
          </size>
         </property>
         <property name="text">
          <string>...</string>
         </property>
         <property name="icon">
          <iconset resource="resource.qrc">
           <normaloff>:/new/img/plus.png</normaloff>:/new/img/plus.png</iconset>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="contactListWidget">
         <property name="modelColumn">
          <number>0</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="convsWidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QListWidget" name="conversationListWidget">
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="loginWidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="3" column="1">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="passwordLine">
         <property name="echoMode">
          <enum>QLineEdit::Password</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="userLabel">
         <property name="text">
          <string>User ID</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="userLine"/>
       </item>
       <item row="5" column="1">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="passwordLabel">
         <property name="text">
          <string>Password</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QPushButton" name="createUserButton">
         <property name="text">
          <string>Create Account</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QPushButton" name="loginButton">
         <property name="text">
          <string>Login</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="networkingWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0,0,1">
       <property name="spacing">
        <number>5</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="serverConfigGroup">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>55</height>
          </size>
         </property>
         <property name="title">
          <string>Chat Server To Use</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="serverAddrLabel">
            <property name="text">
             <string>Server Address:Port</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="serverAddrLine">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>rypi.linuxd.org:19486</string>
            </property>
            <property name="readOnly">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>10</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QCheckBox" name="proxyCB">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>20</height>
          </size>
         </property>
         <property name="text">
          <string>Enable Proxy</string>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="proxySettingsGroup">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>55</height>
          </size>
         </property>
         <property name="title">
          <string>Connect To Proxy Server</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="proxyAddrLabel">
            <property name="text">
             <string>Proxy Address:Port</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="proxyAddrLine">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="readOnly">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="proxyConnectButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Connect To Proxy</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="clientOptionsWidget" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QCheckBox" name="autoSyncCB">
         <property name="text">
          <string>Automatically sync on login</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="enableAdvancedCB">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Enable Advanced Options</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="advancedGroupBox">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="title">
          <string>Advanced Security Options</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QCheckBox" name="setContactPubCB">
            <property name="text">
             <string>Allow manual setting of contacts public key</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="trustDeviceCB">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Trust this device</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="threadsLabel">
            <property name="text">
             <string>Threads to use when creating account: 2</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="threadsSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>16</number>
            </property>
            <property name="pageStep">
             <number>4</number>
            </property>
            <property name="value">
             <number>2</number>
            </property>
            <property name="tracking">
             <bool>true</bool>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="invertedAppearance">
             <bool>false</bool>
            </property>
            <property name="invertedControls">
             <bool>false</bool>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBelow</enum>
            </property>
            <property name="tickInterval">
             <number>2</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
   <zorder>loginWidget</zorder>
   <zorder>contactsWidget</zorder>
   <zorder>convsWidget</zorder>
   <zorder>networkingWidget</zorder>
   <zorder>clientOptionsWidget</zorder>
   <zorder>messagingWidget</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>500</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuChat">
    <property name="title">
     <string>Chat</string>
    </property>
    <addaction name="actionConversations"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <addaction name="actionNetworking"/>
    <addaction name="actionServer_Public_key"/>
    <addaction name="separator"/>
    <addaction name="actionClient"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
    <addaction name="actionDonate"/>
    <addaction name="separator"/>
    <addaction name="actionLicense"/>
    <addaction name="actionCurve_License"/>
    <addaction name="actionQt_License"/>
   </widget>
   <widget class="QMenu" name="menuUser">
    <property name="title">
     <string>User</string>
    </property>
    <addaction name="actionLogin"/>
    <addaction name="actionCreate_Account"/>
    <addaction name="separator"/>
    <addaction name="actionLogout"/>
    <addaction name="actionGet_Public_Key"/>
    <addaction name="actionSync"/>
    <addaction name="actionUpdate_Password"/>
   </widget>
   <widget class="QMenu" name="menuContacts">
    <property name="title">
     <string>Contacts</string>
    </property>
    <addaction name="actionView_Contacts"/>
    <addaction name="actionAdd_Contact"/>
   </widget>
   <addaction name="menuUser"/>
   <addaction name="menuContacts"/>
   <addaction name="menuChat"/>
   <addaction name="menuOptions"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionLogin">
   <property name="text">
    <string>Login To Account</string>
   </property>
  </action>
  <action name="actionCreate_Account">
   <property name="text">
    <string>Create Account</string>
   </property>
  </action>
  <action name="actionView_Contacts">
   <property name="text">
    <string>View Contacts</string>
   </property>
  </action>
  <action name="actionAdd_Contact">
   <property name="text">
    <string>Add Contact</string>
   </property>
  </action>
  <action name="actionConversations">
   <property name="text">
    <string>Conversations</string>
   </property>
  </action>
  <action name="actionNetworking">
   <property name="text">
    <string>Networking</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionDonate">
   <property name="text">
    <string>Donate</string>
   </property>
  </action>
  <action name="actionLicense">
   <property name="text">
    <string>License</string>
   </property>
  </action>
  <action name="actionCurve_License">
   <property name="text">
    <string>curve25519-donna License</string>
   </property>
  </action>
  <action name="actionLogout">
   <property name="text">
    <string>Logout</string>
   </property>
  </action>
  <action name="actionGet_Public_Key">
   <property name="text">
    <string>Get Public Key</string>
   </property>
  </action>
  <action name="actionClient">
   <property name="text">
    <string>Client</string>
   </property>
  </action>
  <action name="actionContactPublicKey">
   <property name="text">
    <string>Get Public Key</string>
   </property>
  </action>
  <action name="actionRemoveContact">
   <property name="text">
    <string>Remove Contact</string>
   </property>
  </action>
  <action name="actionUpdateNickname">
   <property name="text">
    <string>Update Nickname</string>
   </property>
  </action>
  <action name="actionAddToConv">
   <property name="text">
    <string>Add Contact To Conv</string>
   </property>
  </action>
  <action name="actionSetContactPublicKey">
   <property name="text">
    <string>Set Public Key</string>
   </property>
  </action>
  <action name="actionSync">
   <property name="text">
    <string>Sync Account</string>
   </property>
  </action>
  <action name="actionLeaveConv">
   <property name="text">
    <string>Leave Conv</string>
   </property>
  </action>
  <action name="actionGetMembers">
   <property name="text">
    <string>Get Members</string>
   </property>
  </action>
  <action name="actionServer_Public_key">
   <property name="text">
    <string>Server Public key</string>
   </property>
  </action>
  <action name="actionUpdate_Password">
   <property name="text">
    <string>Update Password</string>
   </property>
  </action>
  <action name="actionCreateConvWithContact">
   <property name="text">
    <string>Create Conv With Contact</string>
   </property>
  </action>
  <action name="actionQt_License">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/new/img/Qt Icon.png</normaloff>:/new/img/Qt Icon.png</iconset>
   </property>
   <property name="text">
    <string>Qt License</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
